spec/features/users/dogs/new_spec.rb

```diff
@@ -0,0 +1,111 @@
+require "rails_helper"
+
+RSpec.describe "User Dog Index" do
+  before(:each) do
+    @user = UsersFacade.find_user(1)
+  end
+
+  it 'the user dashboard has an add dog button and it takes you to the add dog path', :vcr do
+    visit user_path(1)
+
+    expect(page).to have_content("Add New Dog")
+    click_button("Add New Dog")
+    expect(current_path).to eq(new_user_dog_path(1))
+  end
+
+  it 'has a form to fill in', :vcr do
+    visit new_user_dog_path(1)
+    
+    expect(page).to have_field("Name")
+    expect(page).to have_field("Breed")
+    expect(page).to have_field("Age")
+    expect(page).to have_select("Size") 
+    expect(page).to have_select("Neutered")
+    expect(page).to have_button("Create Dog")
+
+    fill_in :name, with: "Akila"
+    fill_in :breed, with: "Great Pyrenees"
+    fill_in :age, with: "8"
+    select "large", from: "Size"
+    select "true", from: "Neutered"
+    click_button("Create Dog")
+    
+    expect(current_path).to eq(user_dogs_path(1))
+    expect(page).to have_content("Akila")
+    expect(page).to have_content("Great Pyrenees")
+    expect(page).to have_content(8)
+    expect(page).to have_content("large")
+    expect(page).to have_content("true")
+  end
+
+  describe "#sad-path, dog can not be created if all fields are not filled in" do
```

Comment:
Very thorough sad path testing here with testing every attribute. This looks amazing!

Reply:
Thanks for getting this built out and for your use of thorough testing within the `Dog` new spec!